zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -236,10 +236,14 @@ private[codec] object EncoderDecoder                   {
       if (jsonDecoders.length == 0) {
         ZIO.unit
       } else if (jsonDecoders.length == 1) {
-        jsonDecoders(0)(body).map { result => inputs(0) = result }
+        jsonDecoders(0)(body).map { result => inputs(0) = result }.mapError { err =>
+          HttpCodecError.MalformedBody(err.getMessage())
```

Comment:
I think you should modify `MalformedBody` so that it takes another `Throwable` (or `Option[Throwable]`), so we can attach the original error. Otherwise, we preserve the message, but nothing else, and there could be useful details in that `Throwable`.

Reply:
Thank you for your feedback. I've applied your suggestion and also changed the test case to match the error initial Throwable as a use case example.